examples/svm/plot_svm_kernels.py

```diff
@@ -1,93 +1,279 @@
-    plt.clf()
-
-    plt.scatter(
+def plot_training_data_with_decision_boundary(kernel):
+    # Train the SVC
+    clf = svm.SVC(kernel=kernel, gamma=2).fit(X, y)
+
+    # Settings for plotting
+    _, ax = plt.subplots(figsize=(4, 3))
+    x_min, x_max, y_min, y_max = -3, 3, -3, 3
+    ax.set(xlim=(x_min, x_max), ylim=(y_min, y_max))
+
+    # Plot decision boundary and margins
+    common_params = {"estimator": clf, "X": X, "ax": ax}
+    DecisionBoundaryDisplay.from_estimator(
+        **common_params,
+        response_method="predict",
+        plot_method="pcolormesh",
+        alpha=0.3,
+    )
+    DecisionBoundaryDisplay.from_estimator(
+        **common_params,
+        response_method="decision_function",
+        plot_method="contour",
+        levels=[-1, 0, 1],
+        colors=["k", "k", "k"],
+        linestyles=["--", "-", "--"],
+    )
+
+    # Plot huger circles around samples that serve as support vectors
+    ax.scatter(
         clf.support_vectors_[:, 0],
         clf.support_vectors_[:, 1],
-        s=80,
+        s=250,
         facecolors="none",
-        zorder=10,
         edgecolors="k",
     )
-    plt.scatter(X[:, 0], X[:, 1], c=Y, zorder=10, cmap=plt.cm.Paired, edgecolors="k")
-
-    plt.axis("tight")
-    x_min = -3
-    x_max = 3
-    y_min = -3
-    y_max = 3
-
-    XX, YY = np.mgrid[x_min:x_max:200j, y_min:y_max:200j]
-    Z = clf.decision_function(np.c_[XX.ravel(), YY.ravel()])
-
-    # Put the result into a color plot
-    Z = Z.reshape(XX.shape)
-    plt.figure(fignum, figsize=(4, 3))
-    plt.pcolormesh(XX, YY, Z > 0, cmap=plt.cm.Paired)
-    plt.contour(
-        XX,
-        YY,
-        Z,
-        colors=["k", "k", "k"],
-        linestyles=["--", "-", "--"],
-        levels=[-0.5, 0, 0.5],
-    )
+    # Plot samples by color and add legend
+    ax.scatter(X[:, 0], X[:, 1], c=y, s=150, edgecolors="k")
+    ax.legend(*scatter.legend_elements(), loc="upper right", title="Classes")
+    ax.set_title(f" Decision boundaries of {kernel} kernel in SVC")
+
+    _ = plt.show()
+
+
+# %%
+# Linear kernel
+# *************
+# Linear kernel is the dot products of the input samples:
+#
+# .. math:: K(\mathbf{x}_1, \mathbf{x}_2) = \mathbf{x}_1^\top \mathbf{x}_2
+#
+# It is then applied to any combination of two data points (samples) in the
+# dataset. The dot product of the two points determines the
+# :func:`~sklearn.metrics.pairwise.cosine_similarity` between both points. The
+# higher the value, the more similar the points are.
+plot_training_data_with_decision_boundary("linear")
+
+# %%
+# Training a :class:`~sklearn.svm.SVC` on a linear kernel results in an
+# untransformed feature space and we can see the hyperplane and the margins as
+# straight lines. The support vectors are on the margins. The trained classes do
+# not perfectly capture the data, but thanks to the model's soft margins, we can
+# expect that it has converged, and the position of the classification
+# boundaries has stabilized.
```

Comment:
I moved here the line concerning the choice `C=1` resulting in margins at 1 unit distance, as this seems to be true only for the linear kernel, but it would be nice if you could add a reference or the reason why that is the case.
```suggestion
# Training a :class:`~sklearn.svm.SVC` on a linear kernel results in an
# untransformed feature space where the hyperplane and the margins are straight
# lines. In this case, the choice `C=1` results in margins at 1 unit distance
# from both sides of the hyperplane. Due to the lack of expressivity of the
# linear kernel, the trained classes do not perfectly predict the data.
```